atomicfu/src/nativeMain/kotlin/kotlinx/atomicfu/Synchronized.kt

```diff
@@ -65,6 +65,27 @@ public actual open class SynchronizedObject {
         }
     }
 
+    public fun tryLock(): Boolean {
+        val currentThreadId = pthread_self()!!
+        val state = lock.value
+        if (state.status == UNLOCKED) {
+            val thinLock = LockState(THIN, 1, 0, currentThreadId)
+            if (lock.compareAndSet(state, thinLock)) {
+                return true
+            }
+        } else {
+            if (currentThreadId == state.ownerThreadId) {
+                check(state.status == THIN)
+                check(state.waiters == 0)
+                val nestedLock = LockState(THIN, state.nestedLocks + 1, state.waiters, currentThreadId)
+                if (lock.compareAndSet(state, nestedLock)) {
```

Comment:
It must retry on CAS failure. It cannot return `false` simply due to interference from other thread. It can only return `false` only when the lock is held by another thread.